networkit/test/test_distance_algorithms.py

```diff
@@ -175,12 +183,80 @@ def testMultiTargetSTSP(self):
 					algo = nk.distance.MultiTargetDijkstra(g, source, targets)
 				else:
 					algo = nk.distance.MultiTargetBFS(g, source, targets)
-				algo.setTargets(targets)
-				#algo.setTarget(targets[0])	
+				algo.setSource(source)
+				algo.setTarget(targets[nTargets-1])
+				algo.run()
+				self.assertLessEqual(len(algo.getPredecessors()),g.numberOfNodes())
+				#self.assertLessEqual(len(targets),len(algo.getDistances())) #getdistances -> segfault
+
+	def testSSSP(self):
+		for g in self.genERGraphs():
+			source = nk.graphtools.randomNode(g)
+			for nTargets in [1, 10, 50]:
+				targets = nk.graphtools.randomNodes(g, nTargets)
+				algo = None
+				if g.isWeighted():
+					algo = nk.distance.Dijkstra(g, source, targets, storeNodesSortedByDistance = True)
+				else:
+					algo = nk.distance.BFS(g, source, targets)
 				algo.setSource(source)	
+				algo.setTarget(targets[nTargets-1])
 				algo.run()
-				self.assertLessEqual(len(algo.getPredecessors()), g.numberOfNodes())	
-				self.assertEqual(len(algo.getDistances()), len(targets))
+				self.assertLessEqual(len(algo.getPredecessors(source)),g.numberOfNodes())
+				self.assertLessEqual(len(targets),len(algo.getDistances()))
```

Comment:
I think this test is incorrect, it will always be true since the number of targets (3) is less than the number of nodes of the graph (that is the length of the `getDistances` list). It can be removed.